### PR TITLE
Add --withoutFTIndex (-j) option to zimrecreate tool

### PR DESCRIPTION
**Pull request by data-man**

### CODE REVIEW

**Comment by data-man** [comment]
BTW, `zim::writer::Creator` extracts some Xapian's data to tmp-files even if indexing parameter == false.

**Comment by mgautierfr** [comment]
The option should be named "--noftindex".
It deactivate only the fulltext index, the title index (used for suggestion) is always created.

**Comment by data-man** [comment]
> The option should be named "--noftindex".

Ok.

`do not create a full text index` is the right description?

**Comment by data-man** [comment]
> the title index (used for suggestion) is always created.

How disable it?
I'm currently using `libzim` with `with_xapian=false` option for this.

**Comment by kelson42** [comment]
@data-man @mgautierfr Should be like for `zimwriterfs`:
```
-j, --withoutFTIndex	don't create and add a fulltext index of the content to the ZIM.
```

**Comment by data-man** [comment]
> -j, --withoutFTIndex

Done.

File: src/zimrecreate.cpp [hunk 1]
```diff
@@ -173,14 +173,16 @@ void usage()
...
+    "\t-v, --version           print software version\n"
+    "\t-z, --zstd              use Zstandard as ZIM compression (lzma otherwise)\n"
+    "\t-j, --withoutFTIndex    don't create and add a fulltext index of the content to the ZIM\n";
     return;
 }
 
 int main(int argc, char* argv[])
 {
     bool zstdFlag = false;
+    bool noftindexFlag = false;
```
Comment (mgautierfr) [inline src/zimrecreate.cpp:185]: Please avoid negative boolean.
Use `withFtIndexFlag`.

**Review by mgautierfr — CHANGES_REQUESTED** [review]

**Comment by mgautierfr** [comment]
> > the title index (used for suggestion) is always created.
>
> How disable it?
> I'm currently using libzim with with_xapian=false option for this.

There is no way to do it for now.

**Comment by mgautierfr** [comment]
Please squash all your commits into one and we are good.

**Comment by data-man** [comment]
github-close-pull-request-with-squash

No? :)

**Comment by kelson42** [comment]
@data-man One commit in a merge commit != One commit without merge commit

**Comment by mgautierfr** [comment]
We don't squash the Pull request itself into master.
We always create a merge commit to clearly identify which commits are part of the PR.
There is no option in github to squash all commits in the PR into one but still create a merge commit.

**Comment by data-man** [comment]
I hate git. :(

**Review by mgautierfr — APPROVED** [review]

**Comment by mgautierfr** [comment]
I've tried to be smart and rebase your branch on top of the new master (after the merge of #237) and git was smarter than me.
I messed a bit with the history and github didn't detect the merge.
But it is merged and in master. I'm closing the PR.

**Comment by data-man** [comment]
Thanks!

BTW What if use `docopt` also in zimrecreate?

**Comment by mgautierfr** [comment]
> BTW What if use docopt also in zimrecreate?

Would be a good thing. I haven't mention it as it is not part of this PR. But if you want to change this you are welcomed.

**Comment by data-man** [comment]
> But if you want to change this you are welcomed.

Ok, I'll do!

**Pull request closed without merging**